File: tests/fixtures/py/module_scope/src/nested_function.py
from stripe import StripeClient

client = StripeClient("sk_test")


def outer():
    charge = client.charges.create(amount=100, currency="usd")

    def inner():
        charge = compute_locally()
        return charge.total

    return charge.status, inner


def compute_locally():
    return {"total": 42}

File: tests/fixtures/ts/module_scope/src/nested_function.ts
import Stripe from 'stripe';

const stripe = new Stripe(process.env.STRIPE_KEY!);

export async function outer() {
  const charge = await stripe.charges.create({ amount: 100, currency: 'usd' });

  function inner() {
    const charge = computeLocally();
    return charge.total;
  }

  return { state: charge.status, inner };
}

function computeLocally() {
  return { total: 42 };
}
